File: ABRPOINT.Server/Models/Sanction.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace ABRPOINT.Server.Models;

[Table("sanction")]
[PrimaryKey("Concod","Soccod")]
public partial class Sanction
{
    [Required]
    [Column("concod")]
    [StringLength(10)]
    public string? Concod { get; set; }

    [Required]
    [Column("soccod")]
    [StringLength(2)]
    public string? Soccod { get; set; }

    [Column("empcod")]
    [StringLength(12)]
    public string? Empcod { get; set; }

    [Column("condat", TypeName = "datetime")]
    public DateTime? Condat { get; set; }

    [Column("conjour")]
    [StringLength(1)]
    public string? Conjour { get; set; }

    [Column("condep", TypeName = "datetime")]
    public DateTime? Condep { get; set; }

    [Column("conamdep")]
    [StringLength(1)]
    public string? Conamdep { get; set; }

    [Column("conret", TypeName = "datetime")]
    public DateTime? Conret { get; set; }

    [Column("conamret")]
    [StringLength(1)]
    public string? Conamret { get; set; }

    [Column("abscod")]
    [StringLength(6)]
    public string? Abscod { get; set; }

    [Column("conmotif")]
    [StringLength(50)]
    public string? Conmotif { get; set; }

    [Column("consanc")]
    [StringLength(1)]
    public string? Consanc { get; set; }

    [Column("connbjour")]
    public float? Connbjour { get; set; }

    [Column("conref")]
    [StringLength(20)]
    public string? Conref { get; set; }
}

File: abrpoint.client/src/models/Sanction.ts
export type Sanction = {
    concod?: string;        // N° Ordre
    soccod?: string|null;        // Company Code
    empcod?: string;        // Employee Code
    condat?: Date|null;          // Date of the sanction (congé)
    conjour?: string;       // Full/half day (e.g. 'J' for full day)
    condep?: Date|null;          // Departure Date
    conamdep?: string;      // Departure Morning/Afternoon Indicator (optional)
    conret?: Date|null;          // Return Date
    conamret?: string;      // Return Morning/Afternoon Indicator (optional)
    abscod?: string;        // Absence code (e.g. leave type)
    conmotif?: string;      // Motif (reason for the sanction)
    consanc?: string;       // Sanction type (e.g. warning, dismissal)
    connbjour?: number;     // Number of days
    conref?: string;        // Reference number of the sanction
  }
  